File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/services/ValidatorService.java
package de.vkb.dojo.es.facilityManagement.services;

import org.springframework.stereotype.Component;

import java.util.Optional;

@Component
public class ValidatorService {
    public boolean isValidName(Optional<String> name) {
        return name.map( it -> !it.isBlank() ).orElse(false);
    }
    public boolean isValidName(String name) {
        return isValidName(Optional.ofNullable(name));
    }
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/services/ValidatorService.kt
package de.vkb.dojo.es.humanResources.services

import jakarta.inject.Singleton
import java.util.*

@Singleton
class ValidatorService {
    private val usernamePattern = Regex("^[a-zA-Z0-9-_@.]+$")
    fun isValidUsername(name: Optional<String>): Boolean {
        return name.map { it: String -> it.isNotBlank() && it.matches(usernamePattern) }.orElse(false)
    }

    fun isValidUsername(name: String?): Boolean {
        return isValidUsername(Optional.ofNullable(name))
    }

    fun isValidFullname(name: Optional<String>): Boolean {
        return name.map { it: String -> it.isNotBlank() }.orElse(false)
    }

    fun isValidFullname(name: String?): Boolean {
        return isValidFullname(Optional.ofNullable(name))
    }
}
